libs/wingsdk/src/target-sim/container.inflight.ts

```diff
@@ -126,53 +138,88 @@ export class Container implements IContainerClient, ISimulatorResourceInstance {
       dockerRun.push(a);
     }
 
-    this.log(`starting container from image ${this.imageTag}`);
-    this.log(`docker ${dockerRun.join(" ")}`);
-
-    await shell("docker", dockerRun);
-
-    this.log(`containerName=${this.containerName}`);
+    this.addTrace(
+      `Starting container from ${this.imageTag}...`,
+      TraceType.RESOURCE
+    );
 
-    // wait until the container is running
-    await waitUntil(async () => {
-      const container = JSON.parse(
-        await runCommand("docker", ["inspect", this.containerName])
-      );
-      return container?.[0]?.State?.Running;
+    this.child = this.dockerSpawn("run", dockerRun, {
+      stdout: true,
+      stderr: true,
+      onError: (err) => {
+        this.addTrace(err.stack ?? err.message, TraceType.LOG, LogLevel.ERROR);
+      },
     });
 
-    if (!this.props.containerPort) {
-      return {};
-    }
-
     let hostPort: number | undefined;
     await waitUntil(async () => {
-      const container = JSON.parse(
-        await runCommand("docker", ["inspect", this.containerName])
-      );
+      if (!this.child?.started) {
+        throw new Error(`container ${this.imageTag} stopped unexpectedly`);
+      }
 
-      hostPort =
-        container?.[0]?.NetworkSettings?.Ports?.[
-          `${this.props.containerPort}/tcp`
-        ]?.[0]?.HostPort;
+      const container = await this.tryInspect(this.containerName);
 
-      return hostPort !== undefined;
+      // if we are waiting for a port, check if the container is listening to it
+      if (this.props.containerPort) {
+        hostPort =
+          container?.[0]?.NetworkSettings?.Ports?.[
+            `${this.props.containerPort}/tcp`
+          ]?.[0]?.HostPort;
+
+        return hostPort !== undefined;
+      }
+
+      // if we are not waiting for a port, just check if the container is running
+      return container?.[0]?.State?.Running;
     });
 
-    if (!hostPort) {
-      throw new Error(
-        `Container does not listen to port ${this.props.containerPort}`
-      );
-    }
+    this.addTrace(
+      `Container ${this.imageTag} started`,
+      TraceType.RESOURCE,
+      LogLevel.INFO
+    );
 
     return {
       [HOST_PORT_ATTR]: hostPort,
     };
   }
 
+  private createEnvFile() {
```

Comment:
This method can be async and use the regular fs functions instead of the `std` fs

Reply:
OK, moved to `writeFile` but still prefer to use our mkdtemp because it's nicer.